pkg/ipam/ipam.go

```diff
@@ -107,7 +108,13 @@ type Metadata interface {
 }
 
 // NewIPAM returns a new IP address manager
-func NewIPAM(nodeAddressing types.NodeAddressing, c Configuration, owner Owner, k8sEventReg K8sEventRegister, mtuConfig MtuConfiguration, clientset client.Clientset) *IPAM {
+func NewIPAM(nodeAddressing types.NodeAddressing,
+	c Configuration,
+	owner Owner,
+	k8sEventReg K8sEventRegister,
+	mtuConfig MtuConfiguration,
+	clientset client.Clientset,
+	nodeStore *node.LocalNodeStore) *IPAM {
```

Comment:
I'm not a big fan of introducing this dependency here. 

In particular, we have https://github.com/cilium/cilium/pull/26962 which attempts to decouple some of the dependencies by having both IPAM and BGP rely on a `LocalCiliumNodeResource`.

Given that the list of allocated CIDRs is not coming from the agent IPAM module, but instead is coming from the operator, there is not really a need for this IPAM module to tell BGP or the local node store that we have a new CIDR. Instead, BGP can obtain this information directly from the `LocalCiliumNodeResource` I think, as we already do with the singular pod CIDRs in IPAM kubernetes/clusterpool mode.

Reply:
@gandro thanks for the review. I'll work on refactoring for BGP CP to use `LocalCiliumNodeResource` instead of introducing the `LocalNodeStore` dep.

Reply:
@gandro https://github.com/cilium/cilium/pull/27285 moved to using LocalNodeStore to decouple ourselves from relying on a particular representation of the Node. My intention was to continue the use of LocalNodeStore to keep this decoupled approach. I was under the impression that this was the preferred method moving forward.

Reply:
Commit 0329660 follows the recommendations of https://github.com/cilium/cilium/pull/27100#discussion_r1307355084 by using `LocalCiliumNodeResource` to trigger events instead of the internal `Node` type.